modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingFlowComposable.kt

```diff
@@ -258,9 +265,10 @@ private object OnboardingNavRoute {
         private const val routeBase = "plus_upgrade"
 
         const val sourceArgumentKey = "source"
+        const val showPatronOnlyArgumentKey = "show_patron_only"
         // The route variable should only be used to navigate to the PlusUpgrade screens
         // when they are the startDestination. In all other cases, use the routeWithSource function.
-        const val route = "$routeBase/{$sourceArgumentKey}"
+        const val route = "$routeBase/{$sourceArgumentKey}/{$showPatronOnlyArgumentKey}"
```

Comment:
Thinking about this some more, It seems like my suggestion would break if multiple views were using this view model (probably not an issue in this case, but as a general matter could be a problem) since currently it [is not possible to scope a view model to a particular composable](https://issuetracker.google.com/issues/136967621). 🤔

Reply:
> I've definitely run into this issue of how best to get some data from the Compose layer to the view model, and it doesn't seem like there's a clear best approach. 

That's true. Passing objects via routes from Compose layer to view model is currently [not supported](https://developer.android.com/guide/navigation/navigation-pass-data#supported_argument_types). There are hacks to do so using custom nav type and json string <-> object conversion but it is considered an [anti-pattern](https://stackoverflow.com/a/69060224/193545). I also considered using assisted variables in `HiltViewModel` but as you pointed it is currently not supported for Compose.

> Am I right that we don't actually pass this argument when we navigate, but instead we're just using the default value (and we have some logic to intelligently set the default)? If so, that feels a bit confusing to me.

I understand what you're saying but I wasn't able to find a simpler solution. The approach I took in this PR follows [official guidelines](https://developer.android.com/jetpack/compose/navigation#nav-with-args) to pass nav args similar to how they are passed in "implicit" deep links. Default values are set as arguments for a start destination, the navigation component parses them based on the deep-link-like route that we provide and allows us to access them using `saveStateHandle` in the view model.

> The Navigation component attempts to parse the placeholder values into appropriate types by matching placeholder names to the defined [arguments](https://developer.android.com/guide/navigation/navigation-pass-data#define_destination_arguments) that are defined for the deep link destination.

It is also explained [here](https://stackoverflow.com/a/65544413/193545). Since I require simple boolean and string sources in the view model, sending them using routes fits my case. 

To make it more natural so that the route looks closer to a deep-link-like structure, I made `showPatronOnly` an optional param and added a comment linking the documentation to help understand it in f6f5a8a

```
New: "$routeBase/{$sourceArgumentKey}?{$showPatronOnlyArgumentKey}={$showPatronOnlyArgumentKey}"
Vs 
Old: "$routeBase/{$sourceArgumentKey}/{$showPatronOnlyArgumentKey}"
```

Maybe I should also convert the source string to an optional one. 🤔  

>  thinking about this more because I have the same issue, but I can't use navigation arguments for my use case really (well, I could, but it would be pretty hacky and hard to understand).

The recommended approach to[ getting complex data in view models](https://developer.android.com/jetpack/compose/navigation#retrieving-complex-data) is getting them using the data layer like through a repository. Lmk your use case though I'm not sure if there are good options currently.